src/main/java/seedu/task/logic/parser/FindCommandParser.java

```diff
@@ -32,22 +45,48 @@ public class FindCommandParser implements Parser<FindCommand> {
     public FindCommand parse(String args) throws ParseException {
         requireNonNull(args);
         ArgumentMultimap argMultimap =
-                ArgumentTokenizer.tokenize(args, PREFIX_NAME, PREFIX_DESCRIPTION, PREFIX_TAG);
+                ArgumentTokenizer.tokenize(args, PREFIX_NAME, PREFIX_DESCRIPTION, PREFIX_TAG, PREFIX_ALLMATCH,
+                    PREFIX_DEADLINE, PREFIX_FROM, PREFIX_TO);
 
         if (areTooManyPrefixesPresent(argMultimap, PREFIX_NAME, PREFIX_DESCRIPTION)) {
             throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.MESSAGE_USAGE));
         }
-
         if (argMultimap.getValue(PREFIX_NAME).isPresent()) {
-            return new FindCommand(new NameContainsKeywordsPredicate(argMultimap.getValue(PREFIX_NAME).get()));
+            Set<Name> names = ParserUtil.parseNames(argMultimap.getAllValues(PREFIX_NAME));
+            Name name = ParserUtil.parseName(argMultimap.getValue(PREFIX_NAME).get());
+            if (argMultimap.getValue(PREFIX_ALLMATCH).isPresent()) {
+                return new FindCommand(new NameContainsAllKeywordsPredicate(ParserUtil.parseNamesToList(names)));
+            }
+            return new FindCommand(new NameContainsKeywordsPredicate(name.toString()));
         }
         if (argMultimap.getValue(PREFIX_DESCRIPTION).isPresent()) {
-            return new FindCommand(new DescContainsKeywordsPredicate(argMultimap.getValue(PREFIX_DESCRIPTION).get()));
+            Set<Description> descriptions = ParserUtil.parseDescriptions(argMultimap.getAllValues(PREFIX_DESCRIPTION));
+            Description description = ParserUtil.parseDescription(argMultimap.getValue(PREFIX_DESCRIPTION).get());
+            if (argMultimap.getValue(PREFIX_ALLMATCH).isPresent()) {
+                return new FindCommand(new DescContainsAllKeywordsPredicate(ParserUtil
+                    .parseDescriptionsToList(descriptions)));
+            }
+            return new FindCommand(new DescContainsKeywordsPredicate(description.toString()));
         }
         if (parseTagsForFind(argMultimap.getAllValues(PREFIX_TAG)).isPresent()) {
-            return new FindCommand(new TagsContainsKeywordsPredicate(argMultimap.getAllValues(PREFIX_TAG)));
+            Set<Tag> tags = ParserUtil.parseTags(argMultimap.getAllValues(PREFIX_TAG));
+            if (argMultimap.getValue(PREFIX_ALLMATCH).isPresent()) {
+                return new FindCommand(new TagsContainsAllKeywordsPredicate(ParserUtil.parseTagsToList(tags)));
+            }
+            return new FindCommand(new TagsContainsKeywordsPredicate(ParserUtil.parseTagsToList(tags)));
+        }
```

Comment:
There is quite a lot of code duplication here, possible chance to abstract out to another method? Perhaps it takes in a prefix and returns a find command.

Reply:
created a new parseForFindCommand method to handle this